src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -213,45 +228,44 @@ def stop(self):
 
 
 class AsapoWorker:
-    def __init__(self, config):
-        self.endpoint = config["endpoint"]
-        self.n_threads = config["n_threads"]
-        self.timeout = config.get("timeout", 5)
-        self.beamtime = config["beamtime"]
-        self.token = config["token"]
-        self.ingest_mode = get_ingest_mode(config["ingest_mode"])
-
-        self.data_source = config.get("data_source", None)
-        self.file_regex = config["file_regex"]
-
+    def __init__(self, endpoint, beamtime, token, n_threads, file_regex,
```

Comment:
Shouldn't this also accept a beamline?

Reply:
We are using 'auto' for the beamline.

Reply:
But this is only for tests. In production, the beamline would be given and beamtime would be `'auto'`

Reply:
Done.